packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts

```diff
@@ -258,7 +258,9 @@ export class TaskDefinition extends TaskDefinitionBase {
     if (this.isFargateCompatible && this.networkMode !== NetworkMode.AWS_VPC) {
       throw new Error(`Fargate tasks can only have AwsVpc network mode, got: ${this.networkMode}`);
     }
-
+    if (props.proxyConfiguration && this.networkMode !== NetworkMode.AWS_VPC) {
+      throw new Error(`"proxyConfiguration" requires AwsVpc network mode, got: ${this.networkMode}`);
```

Comment:
nit: I would remove the double quotes around ProxyConfiguration and possibly reword it as 

`ProxyConfiguration can only be used with AwsVpc network mode, got: ${this.networkMode}`